Tests/Shared.Specs/StringAssertionSpecs.cs

```diff
@@ -1785,43 +1785,904 @@ public void When_asserting_string_does_not_end_with_equivalent_of_a_value_and_ac
+        }
+
+        [Fact]
+        public void When_string_containment_more_than_once_is_asserted_and_actual_value_is_null_then_it_should_throw_earlier()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            string actual = null;
+            string expectedSubstring = "XYZ";
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => actual.Should().Contain(expectedSubstring).MoreThan.Once("that is {0}", "required");
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act.Should().Throw<XunitException>().WithMessage(
+#if NETCOREAPP1_1
+                "Expected string <null> to contain \"XYZ\".");
+#else
+                "Expected actual <null> to contain \"XYZ\".");
+#endif
+        }
+
+        #endregion
+
+        #region AtMost
+
+        [Fact]
+        public void When_string_containment_at_most_is_asserted_and_actual_value_contains_the_expected_string_at_most_expected_times_it_should_not_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            string actual = "ABCDEBCDF";
+            string expectedSubstring = "BCD";
+            int expectedTimes = 2;
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => actual.Should().Contain(expectedSubstring).AtMost.Times(expectedTimes);
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_string_containment_at_most_is_asserted_and_actual_value_contains_the_expected_string_but_not_at_most_expected_times_it_should_throw()
```

Comment:
x This is a prime example of why I don't like this augmentation of `Contain`.
"ABCDEF" contains "XYS" 0 times, which is less than 1, but the test fails.

Reply:
The same as in https://github.com/fluentassertions/fluentassertions/pull/962#discussion_r232810763